### PR TITLE
fixed addition of application to folder more than once

### LINKED ISSUE #2916: Able to add the same application in a folder multiple times.
### Version Information
all versions

### What is the expected behaviour?
Should not be able to add the same application in a folder multiple times.

### What is the current behaviour?
Able to add the same application in a folder multiple times

### How to reproduce the issue?
1. Add an app to a folder
2. Add it again
3. Observe the app count of folders are increasing 

### Screenshots or Screencast
[Video](https://www.loom.com/share/cb94c69231ac4bbaa889bed93d19b6f0)

### Any possible solutions?
Back-end validation if app exists in folder

### If the bug is confirmed, would you be willing to submit a PR?
No

### PR DESCRIPTION
**Pull request by asprazz**
Fixes #2916 

Please review and let me know if any changes are required.

## Tests 
- [x] Manually Tested on local
- [x] Test Case written

## Screenshots
test

## Approach
I have used repository to find if record already exists by querieng before creating folderApp record, please let me know 
if any factory method available which can be used to find if app id is already referenced with folder id which can be used.

### CODE REVIEW

**Comment by Navaneeth-pk** [comment]
Please fix the lint errors.
@asprazz

**Comment by asprazz** [comment]
@Navaneeth-pk Please find it done.

File: server/src/services/folder_apps.service.ts [hunk 1]
```diff
@@ -11,6 +11,14 @@ export class FolderAppsService {
   ) {}
 
   async create(folderId: string, appId: string): Promise<FolderApp> {
+    const oldFolderApp = await this.folderAppsRepository.findOne({
```
Comment (gsmithun4) [inline server/src/services/folder_apps.service.ts:14]: shall we rename the variable something like `existingFolderApp`? `old` will be confusing.

File: server/src/services/folder_apps.service.ts [hunk 2]
```diff
@@ -11,6 +11,14 @@ export class FolderAppsService {
...
+      where: { appId, folderId },
+    });
+
+    if (oldFolderApp != null) {
```
Comment (gsmithun4) [inline server/src/services/folder_apps.service.ts:18]: `if (!oldFolderApp)`
will be good, we can capture undefined value also

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: server/src/services/folder_apps.service.ts (hunk 1 above)
Comment (asprazz) [reply server/src/services/folder_apps.service.ts]: Yes! it should be more informative

File: server/src/services/folder_apps.service.ts (hunk 2 above)
Comment (asprazz) [reply server/src/services/folder_apps.service.ts]: I have added the check. Please check.

**Review by gsmithun4 — APPROVED** [review]
LGTM

**Comment by gsmithun4** [comment]
deployed https://folder-bug.herokuapp.com/

**Pull request merged**